### PR TITLE
:bug: Properly set the cluster label name on bootstrap secrets

### LINKED ISSUE #1905: Missing labels for bootstrap secrets
**What steps did you take and what happened:**
I reproduced the behavior using CAPD and CAPA. 
For CAPD, I ran the e2e tests in `test/infrastructure/docker` and inspected the secrets while test was running.
For CAPA, I ran `make create-cluster`.
I then saw the following:
```
$ kubectl get secrets --show-labels
NAME                   TYPE                                  DATA   AGE     LABELS
default-token-bzdd7    kubernetes.io/service-account-token   3      45m     <none>
test2-ca               Opaque                                2      19m     cluster.x-k8s.io/cluster-name=test2
test2-controlplane-0   Opaque                                1      19m     cluster.x-k8s.io/cluster-name=
test2-controlplane-1   Opaque                                1      15m     cluster.x-k8s.io/cluster-name=
test2-controlplane-2   Opaque                                1      15m     cluster.x-k8s.io/cluster-name=
test2-etcd             Opaque                                2      19m     cluster.x-k8s.io/cluster-name=test2
test2-kubeconfig       Opaque                                1      19m     <none>
test2-md-0-5nvnn       Opaque                                1      8m18s   cluster.x-k8s.io/cluster-name=
test2-md-0-nhlqn       Opaque                                1      8m18s   cluster.x-k8s.io/cluster-name=
test2-proxy            Opaque                                2      19m     cluster.x-k8s.io/cluster-name=test2
test2-sa               Opaque                                2      19m     cluster.x-k8s.io/cluster-name=test2


```
I noticed that `test2-kubeconfig` doesn't have any labels and `test2-controlplane-*` and `test2-md-0-*` are missing the value of the `cluster.x-k8s.io/cluster-name` label.

**What did you expect to happen:**
I expect all the resources to be labelled correctly.

**Environment:**

- Cluster-api version: master
- Minikube/KIND version: kind v0.5.1
- Kubernetes version: (use `kubectl version`):
> Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-19T12:36:28Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-20T18:57:36Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}

/kind bug

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1905

### CODE REVIEW

File: bootstrap/kubeadm/controllers/kubeadmconfig_controller_test.go [hunk 1]
```diff
@@ -248,6 +248,10 @@ func TestKubeadmConfigReconciler_Reconcile_MigrateToSecret(t *testing.T) {
 	if string(secret.Data["value"]) != "test" {
 		t.Fatal("expected bootstrap data secret value to match")
 	}
+
+	if secret.Labels[clusterv1.ClusterLabelName] != "cluster" {
+		t.Fatalf("expected bootstrap data secret to have a cluster name label")
```
Comment (wfernandes) [inline bootstrap/kubeadm/controllers/kubeadmconfig_controller_test.go]: We can use `t.Fatal` instead of `t.Fatalf`

Comment (vincepri) [reply bootstrap/kubeadm/controllers/kubeadmconfig_controller_test.go]: I actually wanted to add the formatting, thanks for reminding :)

Comment (vincepri) [reply bootstrap/kubeadm/controllers/kubeadmconfig_controller_test.go]: Done!

**Pull request merged**